toolkit/tools/pkggen/worker/create_worker_chroot.sh

```diff
@@ -7,18 +7,27 @@
 # $3 path to find RPMs. May be in PATH/<arch>/*.rpm
 # $4 path to log directory
 
-[ -n "$1" ] && [ -n "$2" ] && [ -n "$3" ] && [ -n "$4" ] || { echo "Usage: create_worker.sh <./worker_base_folder> <rpms_to_install.txt> <./path_to_rpms> <./log_dir>"; exit; }
+[ -n "$1" ] && [ -n "$2" ] && [ -n "$3" ] && [ -n "$4" ] && [ -n "$5" ] && [ -n "$6" ] || { echo "Usage: create_worker.sh <./worker_base_folder> <rpms_to_install.txt> <./path_to_rpms> <./log_dir> <./bldtracker> <./timestamp_output_directory>"; exit; }
 
 chroot_base=$1
 packages=$2
 rpm_path=$3
 log_path=$4
+bldtracker=$5
+timestamp_dir=$6
 
 chroot_name="worker_chroot"
 chroot_builder_folder=$chroot_base/$chroot_name
 chroot_archive=$chroot_base/$chroot_name.tar.gz
 chroot_log="$log_path"/$chroot_name.log
 
+$bldtracker \
+    --script-name="create_worker_chroot.sh" \
```

Comment:
Please avoid repetitions (see: [the DRY principle](https://en.wikipedia.org/wiki/Don%27t_repeat_yourself)). You can save the script name to a variable and re-use it everywhere. Also, since the script name may change, I'd for instance suggest `script_name=$(basename $0)"` to set the variable. The value of `$0` should be the path used to run the script and `basename` will extract the file name.

Reply:
Closer but we're still re-running `basename` every single time we call the tool - let's just save the value once and re-use it elsewhere.